### PR TITLE
Create a Mesh from an existing lattice

### PR DESCRIPTION
**Pull request by tjlaboss**
I often find myself doing mesh tallies on a lattice. Here's a class method for `Mesh()` to streamline the process.

### CODE REVIEW

File: openmc/mesh.py [hunk 1]
```diff
@@ -182,6 +182,40 @@ def from_hdf5(cls, group):
...
+        ----------
+        lattice : openmc.RectLattice
+            Rectangular lattice used as a template for this mesh
+        division : int, optional
+            Number of mesh cells per lattice cell.
+            If not specified, there will be 1 mesh cell per lattice cell.
+        mesh_id : int, optional
+            Unique identifier for the mesh
+        name : str, optional
+            Name of the mesh
```
Comment (smharper) [inline openmc/mesh.py]: Remove the `, optional` from these three arguments.

**Comment by tjlaboss** [comment]
@paulromano , I added a unit test. I will need some guidance if this isn't quite right.

**Comment by tjlaboss** [comment]
Well, that didn't work, apparently because I renamed `rlat2` --> `rlat2d`

https://travis-ci.org/mit-crpg/openmc/jobs/382368858#L1720

```python
==================================== ERRORS ====================================
________________________ ERROR at setup of test_mesh2d _________________________
file /home/travis/build/mit-crpg/openmc/tests/unit_tests/test_mesh_from_lattice.py, line 89
  def test_mesh2d(rlat2d):
E       fixture 'rlat2d' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cell_with_lattice, cov, doctest_namespace, monkeypatch, pincell1, pincell2, pytestconfig, record_property, record_xml_attribute, record_xml_property, recwarn, rlat2, rlat3, run_in_tmpdir, sphere_model, tmpdir, tmpdir_factory, uo2, water, zr
>       use 'pytest --fixtures [testpath]' for help on them.
/home/travis/build/mit-crpg/openmc/tests/unit_tests/test_mesh_from_lattice.py:89
________________________ ERROR at setup of test_mesh3d _________________________
file /home/travis/build/mit-crpg/openmc/tests/unit_tests/test_mesh_from_lattice.py, line 104
  def test_mesh3d(rlat3d):
E       fixture 'rlat3d' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cell_with_lattice, cov, doctest_namespace, monkeypatch, pincell1, pincell2, pytestconfig, record_property, record_xml_attribute, record_xml_property, recwarn, rlat2, rlat3, run_in_tmpdir, sphere_model, tmpdir, tmpdir_factory, uo2, water, zr
>       use 'pytest --fixtures [testpath]' for help on them.
```

**Comment by tjlaboss** [comment]
7801761a7035f3ba2f8a326390b6fbf9fda87f5a seems to have fixed it, knock on wood.

File: tests/unit_tests/test_mesh_from_lattice.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    ]
+
+    return lattice
+
+
+def test_mesh2d(rlat2):
+    shape = np.array(rlat2.shape)
+    width = shape*rlat2.pitch
+    
+    mesh1 = openmc.Mesh().from_rect_lattice(rlat2)
```
[the remark this answers is not in the document]
Comment (tjlaboss) [reply tests/unit_tests/test_mesh_from_lattice.py]: TIL you can do that.

**Comment by paulromano** [comment]
@smharper Are you good with this PR now?

**Comment by smharper** [comment]
Looks good to me!  Merging.

**Pull request merged**